src-plugins/iterativeClosestPoint/iterativeClosestPointToolBox.cpp

```diff
@@ -164,48 +150,57 @@ bool iterativeClosestPointToolBox::registered()
     {
-        return;
-    }
-
-    dtkSmartPointer <dtkAbstractProcess> process = dtkAbstractProcessFactory::instance()->createSmartPointer("iterativeClosestPointProcess");
-
-    medAbstractData *sourceData = d->currentView->layerData(d->layerSource->currentIndex()-1);
-    medAbstractData *targetData = d->currentView->layerData(d->layerTarget->currentIndex()-1);
+        medAbstractData *sourceData = d->currentView->layerData(d->layerSource->currentIndex()-1);
+        medAbstractData *targetData = d->currentView->layerData(d->layerTarget->currentIndex()-1);
     
-    if (!targetData || !sourceData)
-    {
-        return;
+        if (targetData && sourceData)
+        {
+            d->process = new iterativeClosestPointProcess();
+            d->process->setInput(sourceData,  0);
+            d->process->setInput(targetData, 1);
+
+            d->process->setParameter((double)d->bStartByMatchingCentroids->isChecked(),0);
+            d->process->setParameter((double)d->bTransformationComboBox->currentIndex(),1);
+            d->process->setParameter((double)d->bCheckMeanDistance->isChecked(),2);
+            d->process->setParameter(d->MaxMeanDistance->value(),3);
+            d->process->setParameter((double)d->MaxNumIterations->value(),4);
+            d->process->setParameter((double)d->MaxNumLandmarks->value(),5);
+            d->process->setParameter(d->ScaleFactor->value(),6);
+
+            // Progression stack
+            medRunnableProcess *runProcess = new medRunnableProcess;
+            runProcess->setProcess (d->process);
+            d->progression_stack->addJobItem(runProcess, "Progress:");
+            d->progression_stack->disableCancel(runProcess);
+
+            connect (runProcess, SIGNAL (success   (QObject*)),    this, SIGNAL (success ()));
+            connect (runProcess, SIGNAL (success   (QObject*)),    this, SLOT   (displayOutput()));
+            connect (runProcess, SIGNAL (failure   (QObject*)),    this, SIGNAL (failure ()));
+            connect (runProcess, SIGNAL (failure   (int)),         this, SLOT   (handleDisplayError(int)));
+            connect (runProcess, SIGNAL (cancelled (QObject*)),    this, SIGNAL (failure ()));
+            connect (runProcess, SIGNAL (activate(QObject*,bool)), d->progression_stack, SLOT(setActive(QObject*,bool)));
+
+            medJobManager::instance()->registerJobItem(runProcess);
+            QThreadPool::globalInstance()->start(dynamic_cast<QRunnable*>(runProcess));
+        }
+        else
+        {
+            medMessageController::instance()->showError(tr("Choose two meshs"),3000);
+        }
     }
-
-    iterativeClosestPointProcess *process_Registration = dynamic_cast<iterativeClosestPointProcess *>(process.data());
-    if (!process_Registration)
+    else
     {
-        qWarning() << "iterativeClosestPointProcess process doesn't exist" ;
-        return;
+        medMessageController::instance()->showError(tr("Drop two meshs in the view"),3000);
```

Comment:
ah, it's interesting because i saw both grammar for meshs/meshes. Do you all agree with "meshes"?